esmvalcore/cmor/check.py

```diff
@@ -402,13 +403,30 @@ def _check_time_coord(self):
             self.report_error(self._does_msg, var_name,
                               'have time reference units')
         else:
+            old_units = coord.units
             coord.convert_units(
                 cf_units.Unit(
-                    'days since 1950-1-1 00:00:00',
+                    'days since 1850-1-1 00:00:00',
                     calendar=coord.units.calendar))
             simplified_cal = self._simplify_calendar(coord.units.calendar)
             coord.units = cf_units.Unit(coord.units.origin, simplified_cal)
 
+            attrs = cube.attributes
+            if 'branch_time_in_child' in attrs:
+                attr = 'branch_time_in_child'
+                value = attrs[attr]
+                cube.attributes[attr] = old_units.convert(value, coord.units)
```

Comment:
```suggestion
                attrs[attr] = old_units.convert(attrs[attr], coord.units)
```
would be a bit shorter